### PR TITLE
stm32f3: Add support for OPAMP

### PR DESCRIPTION
**Pull request by ben-brewer-codethink**
Added (and tested) OPAMP support for STM32F3.

### CODE REVIEW

**Comment by ben-brewer-codethink** [comment]
Fixed some accessor functions.

File: lib/stm32/f3/opamp.c [hunk 1]
```diff
@@ -0,0 +1,156 @@
...
+
+void opamp_vps_select(uint32_t base, uint32_t vps)
+{
+	OPAMP_CSR(base) &= ~(OPAMP_CSR_VPS_SEL_MASK
+		<< OPAMP_CSR_VPS_SEL_SHIFT);
+	OPAMP_CSR(base) |= vps << OPAMP_CSR_VPS_SEL_SHIFT;
+}
+
+void opamp_vms_select(uint32_t base, uint32_t vms)
+{
```
Comment (karlp) [inline lib/stm32/f3/opamp.c:149]: why _select on all of these, with _set for others?

Comment (ben-brewer-codethink) [reply lib/stm32/f3/opamp.c:149]: I used select wherever the input was expected to be an enum, and set where it's a value. Saw that kind of thing elsewhere and assumed, but am happy to change them all to set if that makes it more consistent?

**Comment by karlp** [comment]
I've pulled this locally, but things that would make it "way more rad" are:
* doxygen of any sort for the functions and their arguments
* doxygen grouping of the registers and values

(I'm also 99% sure we're going to have to move this code into common/opamp for l1, l4, and l5, but... let'ss make that tomorrow's problem, not todays :)

**Comment by karlp** [comment]
hah! I see now you were the one who added this to f3!  you knew you had a copy in g4!  naughty :)

Anyway, as this was "first" it has landed in master now, thanks!

**Pull request closed without merging**